### PR TITLE
MultiPartConfig in HTTP Vert.x extension is insufficiently documented

### LINKED ISSUE #30046: MultiPartConfig in HTTP Vert.x extension is insufficiently documented
### Describe the bug

Hello,

I have one opinion (you can ignore that one) and one concern (IMO bug) regarding https://github.com/quarkusio/quarkus/pull/29729 documentation.

**Opinion**: Please document this feature in more detail and ideally provide an example in docs. Documentation does not provide single example on `MultiPartConfig` and `fileContentTypes`. I understand you (and me) know what `MultiPartConfig#fileContentType` should be used for because we read this PR, but users didn't not, they only see this configuration property with text `A list of {@code ContentType} to indicate whether a given multipart field should be handled as a file part.`
With context of #29729 that sentence is completely clear, but you can only find this in configuration properties list.

**Concern** This PR added `MultiPartConfig` to Vert.x HTTP extension. `MultiPartConfig` only used by RESTEasy Reactive and ignored by every other extension (RESTEasy Classic, Reactive Routes, ...) while that fact is literally nowhere mentioned. 
 How is it possible to have the configuration in Vert.X HTTP extension and do not mention it only works for RESTEasy Reactive? How should user know that?

### Expected behavior

one of:
- I expect `MultiPartConfig#fileContentType` configuration property works for RESTEasy Classic, Reactive Routes etc.
- Document that it only works for RESTEasy Reactive

### Actual behavior

- only works for RESTEasy Reactive
- docs don't tell me that the property only works for single extension

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by pedroigor**
Closes #30046

### CODE REVIEW

**Comment by michalvavrik** [comment]
The second sentence clarifies config property usage, thanks. I think you should not use phrase comma-separated list as strictly speaking if you use YAML you can define list without commas and you don't just write docs for application.properties; but I can see this phrase used in Quarkus multiple times, so...

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/MultiPartConfig.java [hunk 1]
```diff
@@ -15,7 +15,12 @@
 public class MultiPartConfig {
 
     /**
-     * A list of {@code ContentType} to indicate whether a given multipart field should be handled as a file part.
+     * A comma-separated list of {@code ContentType} to indicate whether a given multipart field should be handled as a file
+     * part.
+     *
+     * You can use this setting to force HTTP-based extensions to parse a message part as a file based on its content type.
+     *
+     * For now, this setting only works if the Resteasy Reactive extension is installed.
```
Comment (gsmet) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/MultiPartConfig.java:23]:
```suggestion
     * For now, this setting only works when using RESTEasy Reactive.
```

**Review by gsmet — COMMENTED** [review]
I suggested a small change.

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/MultiPartConfig.java (hunk 1 above)
Comment (pedroigor) [reply extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/MultiPartConfig.java]: Thanks, done.

**Review by cescoffier — APPROVED** [review]

**Pull request merged**